src/components/IncidentDetails/DefinitionOfDone/EvidenceView/index.tsx

```diff
@@ -0,0 +1,277 @@
+        const component = data[0];
+        setComp(component);
+        setLoading(false);
+      })
+      .catch((err) => {
+        console.log(err);
+        setLoading(false);
+      });
+  }, [evidence]);
+
+  if (loading) {
+    return <Loading text={`Loading component please wait...`} />;
+  }
+
+  if (!comp || !evidence.component_id) {
+    return null;
+  }
+
+  return (
+    <div className={clsx("overflow-hidden p-2", className)} {...rest}>
+      <span>
+        <Icon className="inline" name={comp.icon} size="xl" /> {comp.name}
+      </span>
+    </div>
+  );
+}
+
+function ConfigEvidence({
+  evidence,
+  size,
+  className,
+  ...rest
+}: EvidenceViewProps) {
+  const [config, setConfig] = useState<any>();
+  const { loading, setLoading } = useLoader();
+
+  useEffect(() => {
+    if (!evidence?.config_id) {
+      return;
+    }
+    setLoading(true);
+    getConfig(evidence.config_id)
+      .then(({ data }) => {
+        setConfig(data[0]);
+        setLoading(false);
+      })
+      .catch((err) => {
+        console.log(err);
+        setLoading(false);
+      });
+  }, [evidence]);
+
+  if (loading) {
+    return <Loading text={`Loading config please wait...`} />;
+  }
+
+  if (!config || !evidence.config_id) {
+    return null;
+  }
+
+  return (
+    <div className={clsx("overflow-hidden p-2", className)} {...rest}>
+      <Icon
+        name={config.external_type}
+        secondary={config.config_type}
+        size="lg"
+      />{" "}
+      <span className="pl-1"> {config.name} </span>{" "}
+    </div>
+  );
+}
+
+function HealthEvidence({
+  evidence,
+  size,
+  className,
+  ...rest
+}: EvidenceViewProps) {
+  const [check, setCheck] = useState<any>();
+
+  useEffect(() => {
+    const healthEvidence: any = evidence.evidence;
+    const id = healthEvidence.check_id;
+    const includeMessages = healthEvidence.includeMessages;
+    const start = healthEvidence.start;
+    fetchCheckDetails(id, start, includeMessages);
+  }, [evidence]);
+
+  const fetchCheckDetails = (
+    id: string,
+    start: string,
+    includeMessages: boolean
+  ) => {
+    const payload = {
+      check: id,
+      includeMessages,
+      start
+    };
+    getCanaries(payload).then((results: any) => {
+      if (results == null || results.data.checks.length === 0) {
+        return;
+      }
+      setCheck(results.data.checks[0]);
+    });
+  };
+
+  return (
+    <div className="overflow-hidden p-2" {...rest}>
+      <div className={`flex flex-row`} {...rest}>
+        <div className={clsx("flex-shrink-0", "pr-2")}>
+          <Icon name={check?.icon || check?.type} size={"sm"} />
+        </div>
+        <div className={clsx("overflow-hidden")}>
+          <div className="flex flex-row">
+            <span
+              title={check?.name}
+              className={clsx(
+                "text-gray-800 font-semibold whitespace-nowrap overflow-ellipsis overflow-hidden pr-4"
+              )}
+            >
+              {check?.name}
```

Comment:
Why is this duplicayed ?

Reply:
I have created a simpler version here as the regular evidence views are heavy to render because the view contains code for other components and their supporting logic.